src/types.ts

```diff
@@ -58,15 +60,15 @@ interface Options {
   /**
    * Extend route records
    */
-  extendRoute?: (route: any, parent: any | undefined) => any | void
+  extendRoute?: (route: PrepareRoute, parent: PrepareRoute | undefined) => Route | void
   /**
    * Custom generated routes
    */
-  onRoutesGenerated?: (routes: any[]) => any[] | void | Promise<any[] | void>
+  onRoutesGenerated?: (routes: PrepareRoute[]) => Route[] | Promise<Route[] | void>
   /**
    * Custom generated client code
    */
-  onClientGenerated?: (clientCode: string) => string | void | Promise<string | void>
+  onClientGenerated?: (clientCode: string) => string | Promise<string | void>
```

Comment:
We don't have consistent field in route to let typescript know what type is this route, so currently we can't resolve these types, you can check generate test file, there is a lot of typescript errors, that is why I keep it `any`.
But maybe we can export these types for users.

Reply:
You are right